src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -655,9 +655,9 @@ private static int getTypeDeclarationNameMatchingCountDiff(String outerTypeDeclN
                                                                TypeDeclDesc firstTypeDecl,
                                                                TypeDeclDesc secondTypeDecl) {
         int diff = Integer.compare(
-            CheckUtil.typeDeclarationNameMatchingCount(
+            CheckUtil.getAnonSuperTypeMatchingCount(
```

Comment:
this change is weird, naming is confusing. 
host method is ` getTypeDeclarationNameMatchingCountDiff` and whole logic is in `getAnonSuperTypeMatchingCount`

why "AnnonSuper"  is so critical ?

it make sense for registerAnonymousInnerClassToSuperClass, but not in this context.
looks like getAnonSuperTypeMatchingCount name is root of confusion.

Reply:
The matching count way is different for anon inner classes and regular classes, that is why this is specifically mentioned. 
For regular classes we follow:
```java
/**                                                                                    
 * Calculates and returns the type declaration name matching count.                    
 *                                                                                     
 * <p>                                                                                 
 * Suppose our pattern class is {@code foo.a.b} and class to be matched is             
 * {@code foo.a.ball} then type declaration name matching count would be calculated by 
 * comparing every character, and updating main counter when we hit "." to prevent matc
 * "a.b" with "a.ball". In this case type declaration name matching count              
 * would be equal to 6 and not 7 (b of ball is not counted).                           
 * </p>                                                                                
 *                                                                                     
 * @param patternClass class against which the given class has to be matched           
 * @param classToBeMatched class to be matched                                         
 * @return class name matching count                                                   
 */                                                                                    
private static int typeDeclarationNameMatchingCount(String patternClass,               
                                                   String classToBeMatched) {          
    final int length = Math.min(classToBeMatched.length(), patternClass.length());     
    int result = 0;                                                                    
    for (int idx = 0;                                                                  
         idx < length && patternClass.charAt(idx) == classToBeMatched.charAt(idx);     
         ++idx) {                                                                      
        if (patternClass.charAt(idx) == PACKAGE_SEPARATOR.charAt(0)) {                 
            result = idx;                                                              
        }                                                                              
    }                                                                                  
    return result;                                                                     
}                                                                                      
```

Reply:
it is ok to have different methods for each type of strucutre, but why 
`getTypeDeclarationNameMatchingCountDiff` is not
`getAnnonTypeDeclarationNameMatchingCountDiff` ?

as we use `CheckUtil.getAnonSuperTypeMatchingCount` in it.

Reply:
Changed to `getAnonTypeDeclarationNameMatchingCountDiff`, done.